numpy/lib/npyio.py

```diff
@@ -1071,6 +1071,15 @@ def read_data(chunk_size):
 
     dtype_types, packing = flatten_dtype_internal(dtype)
 
+    if usecols is not None:
+        if len(dtype_types) > 1 and len(usecols) != len(dtype_types):
```

Comment:
perhaps clearer as:
```suggestion
        if len(dtype_types) <= 1:
            pass  # this is ok because ???
        elif len(usecols) != len(dtype_types):
```

Reply:
although i understand that this should be a rare case, maybe makes sense to make it work for len(dtype_types) >= 1 , for one element structured dtypes ?